### ISSUE TITLE
Advanced file type detection for preview irrespective of file extension

### ISSUE THREAD

**Issue by PhantomLord72** [kartik-v/bootstrap-fileinput#1792]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest `master` branch of bootstrap-fileinput.
- [X] This is not an usage question. I confirm having read the plugin [documentation](http://plugins.krajee.com/file-input) and [demos](http://plugins.krajee.com/file-input/demo).
- [ ] This is not a general programming / coding question. (Those should be directed to the [webtips Q & A forum](http://webtips.krajee.com/questions)).
- [X] I have attempted to find the simplest possible steps to reproduce the issue.
- [ ] I have included a failing test as a pull request (Optional).

## Steps to reproduce the issue

1. load some demo from http://plugins.krajee.com/file-input/demo, even the basic demo is not working
2. load a pdf without extension
3. click on the upload button

## Expected behavior and actual behavior
When I follow those steps, I see... the preview show binary contents

I was expecting... the fileinput analyze the mimetype based on its contents header of the file and show the corresponding preview

## Environment

Browsers

- [X] Google Chrome
- [ ] Mozilla Firefox
- [ ] Internet Explorer
- [ ] Safari

Operating System

- [X] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] Mobile

Libraries

- jQuery version: 3.6.0
- bootstrap-fileinput version: 5.2.7

## Isolating the problem

- [X] This bug happens [on the plugin demos page](http://plugins.krajee.com/file-input/demo)
- [ ] The bug happens consistently across all tested browsers
- [ ] This bug happens when using bootstrap-fileinput without other plugins
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

**Comment by kartik-v** [comment]
New enhancements added with above commit.

 - New property `autoRenameExtension` which defaults to `true` - this will auto detect mime type and append the parsed new extension.
- Load the following new plugin files before `fileinput.min.js` - this is needed to auto parse the mime type based on file contents.
```html
<!-- buffer.min.js and filetype.min.js are necessary in the order listed for advanced mime type parsing and more correct
     preview. This is a feature available since v5.2.9 and is needed if you want to ensure file mime type is parsed 
     correctly even if the local file's extension is named incorrectly. This will ensure more correct preview of the
     selected file (note: this will involve a small processing overhead in scanning of file contents locally). -->
<script src="https://cdn.jsdelivr.net/gh/kartik-v/bootstrap-fileinput@5.2.9/js/plugins/buffer.min.js" type="text/javascript"></script>
<script src="https://cdn.jsdelivr.net/gh/kartik-v/bootstrap-fileinput@5.2.9/js/plugins/filetype.min.js" type="text/javascript"></script>
```

**Comment by PhantomLord72** [comment]
This is a great enhancement. however it does not work yet:
Putting breakpoint on line 5674:

5671	if (typeof FileTypeParser !== "undefined") {
5672		fileData = new Uint8Array(theFile.target.result);
5673		new FileTypeParser().parse(fileData).then(function (result){
5674			processPreview(result && result.mime || mime, result && result.ext || '');
5675		});

variables contains these values:
result.ext = "pdf"
result.mime = "application/pdf"
mime = ""

and the preview does not update with the correct mimetype, in this case is a plain "pdf".
Please help me to troubleshoot this bug
